### PR TITLE
[WIP] Try to get rendering working on Android

### PR DESCRIPTION
**Pull request by torkleyy**
I fixed all the compile errors and crashes.. and in return Android now gives me a black screen. I'm not quite sure how to debug it :(

cc @ozkriff Maybe you have an idea since you're building your games for Android?

---
This change is [Reviewable](https://reviewable.io/reviews/amethyst/amethyst/565)

### CODE REVIEW

**Comment by anderejd** [comment]
I've had some fun with black screens on android. My solution was spamming my code with glGetError for every single opengl call. Should those calls have been there from the start? Possibly. There are most likely better ways to deal with that though :)
Have you tried using the android emulator?
Did a quick googling and found this: https://github.com/google/gapid

**Comment by Xaeroxe** [comment]
Impressive!!  I didn't know we were even this close.  I'll look into helping out here once I finish the specs PR and my animation PR.

File: amethyst_renderer/src/renderer.rs [hunk 1]
```diff
@@ -275,13 +275,13 @@ fn init_backend(wb: WindowBuilder, el: &EventsLoop, config: &DisplayConfig) -> R
...
 fn init_backend(wb: WindowBuilder, el: &EventsLoop, config: &DisplayConfig) -> Result<Backend> {
     use gfx_window_glutin as win;
-    use glutin::{self, GlProfile, GlRequest};
+    use glutin::{self, Api, GlProfile, GlRequest};
 
     let ctx = glutin::ContextBuilder::new()
         .with_multisampling(config.multisampling)
         .with_vsync(config.vsync)
         .with_gl_profile(GlProfile::Core)
-        .with_gl(GlRequest::Latest);
```
Comment (Rhuagh) [inline amethyst_renderer/src/renderer.rs]: This should be a separate backend imho.

File: ametrine/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[package]
```
Comment (Rhuagh) [inline ametrine/Cargo.toml]: This looks like an example, maybe put it under the examples instead of a free standing crate?

File: src/app.rs [hunk 3]
```diff
@@ -315,7 +315,7 @@ impl<'a, 'b, T> ApplicationBuilder<'a, 'b, T> {
                     message
                 ))
             })
-            .level(LevelFilter::Debug)
+            .level(LevelFilter::Warn)
```
Comment (Rhuagh) [inline src/app.rs]: We should make this controllable from the commandline (you can ignore it in this PR though).

File: amethyst_renderer/src/pass/shaders/vertex/basic.glsl [hunk 4]
```diff
@@ -1,6 +1,4 @@
-// TODO: Needs documentation.
-
-#version 150 core
+#version 300 es
```
Comment (Rhuagh) [inline amethyst_renderer/src/pass/shaders/vertex/basic.glsl]: Will this still work on a standard desktop?

**Comment by torkleyy** [comment]
These changes are not intended to be merged like this, I was just editing the code until it worked :P

**Comment by Rhuagh** [comment]
Ok :)

File: ametrine/Cargo.toml (hunk 2 above)
Comment (torkleyy) [reply ametrine/Cargo.toml]: I'm not sure if it can be an example, depends on whether `cargo-apk` supports that.

**Comment by torkleyy** [comment]
Okay, I've tried to debug the whole thing, but didn't find something yet. Here's the gfxtrace if somebody wants to inspect it. I'm currently only getting some black lines :(

[ametrine_20180210_1107.gfxtrace.zip](https://github.com/amethyst/amethyst/files/1713249/ametrine_20180210_1107.gfxtrace.zip)

**Comment by torkleyy** [comment]
This will be done for hal instead.

**Pull request closed without merging**